### PR TITLE
Update quick starts to encourage local development

### LINKED ISSUE #3278: Create the "build your own environment" quick start
This new topic follows the interactive quick starts. Readers will learn to install VS Code and .NET Core. They will validate the installation by creating a first console application.

### LINKED ISSUE #3563: Update C# Quick Starts for online and local experience.
The C# Quick Starts currently focus on the online experience, and don't provide instructions for developers that are working locally.

Consider this plan:  Each online quick start would be duplicated by a local topic.  Where both exist, only the online quick start appears in the TOC.  The start of each online quick start will include a note to direct users to the local version. Each local version will link to the next local quick start.

### PR DESCRIPTION
**Pull request by BillWagner**
Fixes #3563 
Fixes #3278 

This PR creates a parallel track through the quick starts where a reader can run all the code on their own machine.

The reader is directed to the new .NET Core download page when ready to install .NET Core and continue locally.

/cc @rowanmiller @LadyNaggaga

### CODE REVIEW

File: docs/csharp/quick-starts/branches-and-loops-local.md [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+
+```csharp
+int b = 3;
+```
+
+Type `dotnet run` again. Because the answer is less than 10, nothing is printed. The **condition** you're testing is false. You don't have any code to execute because you've only
+written one of the possible branches for an `if` statement: the true branch.
+
+> [!TIP]
+> As you explore C# (or any programming language), you'll
```
Comment (Rick-Anderson) [inline docs/csharp/quick-starts/branches-and-loops-local.md]: After responding more than a hundred times to  livefyre with:
Wow, awesome error message!  - I'd have to write the preceding as:

 As you explore programming language, you'll
  make mistakes coding mistake. The **compiler** will
  find and report the errors. Look closely the the error output and the code that generated the error. The compler error can usually help you find the problem.

File: docs/csharp/quick-starts/branches-and-loops-local.md [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+> make mistakes when you write code. The **compiler** will
+> find those errors and report them to you. When the output
+> contains error messages, look closely at the example code,
+> and the code in the interactive window to see what to fix.
+> That exercise will help you learn the structure of C# code.     
+
+This first sample shows the power of `if` and boolean types. A *boolean* is a variable that can have one of two values: `true` or `false`. C# defines a special type, `bool` for boolean variables. The `if` statement checks the value of a `bool`. When the value is `true`, the statement following the `if` executes. Otherwise, it is skipped. 
+
+This process of checking conditions and executing statements based on those conditions is very powerful. Let's explore more
```
Comment (Rick-Anderson) [inline docs/csharp/quick-starts/branches-and-loops-local.md]: ~Let's explore more~

File: docs/csharp/quick-starts/branches-and-loops-local.md [hunk 3]
```diff
@@ -0,0 +1,305 @@
...
+        {
+            ExploreIf();
+        }
+    }
+}
+```
+
+## Use loops to repeat operations
+
+Comment out the call to `ExploreIf()`. It will make the output less cluttered as you work in this section:
```
Comment (Rick-Anderson) [inline docs/csharp/quick-starts/branches-and-loops-local.md]: I would move all the non-loop info to the preceding section. Anyone jumping to this H2 will be confused.

File: docs/csharp/quick-starts/branches-and-loops-local.md [hunk 4]
```diff
@@ -0,0 +1,305 @@
...
+
+```csharp
+//ExploreIf();
+```
+
+The `//` starts a **comment** in C#. Comments are any text you want to keep in your source code, but not execute as code. The compiler does not generate any executable code from comments.
+
+Another important concept to create larger programs is **loops**. You'll
```
Comment (Rick-Anderson) [inline docs/csharp/quick-starts/branches-and-loops-local.md]: I'd start here  but remove ~Another important concept to create larger programs is~
I avoid another as there is an infinite anothers - it's dead wood. 
No need to tell them it's important. If it wasn't important, you wouldn't be explaining it.  The statement is incorrect - loops are important to small programs too. I'd just tell them about loops.

In this section you use loops to  repeat statements ~that you want executed more than once~

File: docs/csharp/quick-starts/branches-and-loops-local.md [hunk 5]
```diff
@@ -0,0 +1,305 @@
...
+executing those statements until the condition is false.
+
+There's one other new operator in this example. The `++` after
+the `counter` variable is the **increment** operator. It adds 1
+to the value of counter, and stores that value in the counter variable.
+
+> [!IMPORTANT]
+> Make sure that the `while` loop condition does switch to
+> false as you execute the code. Otherwise, you create an
+> **infinite loop** where your program never ends. Let's
```
Comment (Rick-Anderson) [inline docs/csharp/quick-starts/branches-and-loops-local.md]: ASP.NET avoid let us - prefer
That is not demonstrated in this sample

File: docs/csharp/quick-starts/branches-and-loops-local.md [hunk 6]
```diff
@@ -0,0 +1,305 @@
...
+> not demonstrate that, because you have to force your program to
+> quit using **CTRL-C** or other means.
+
+The `while` loop tests the condition before executing the code
+following the `while`. The `do` ... `while` loop executes the 
+code first, and then checks the condition. It looks like this:
```
Comment (Rick-Anderson) [inline docs/csharp/quick-starts/branches-and-loops-local.md]: ~It looks like this:~
The  do while loop is shown in the following code:

File: docs/csharp/quick-starts/branches-and-loops-local.md [hunk 7]
```diff
@@ -0,0 +1,305 @@
...
+do
+{
+  Console.WriteLine($"Hello World! The counter is {counter}");
+  counter++;
+} while (counter < 10);
+```
+
+This `do` loop and the earlier `while` loop work the same. 
+
+Let's move on to one last loop statement.
```
Comment (Rick-Anderson) [inline docs/csharp/quick-starts/branches-and-loops-local.md]: ~Let's move on to one last loop statement.~

dead wood

File: docs/csharp/quick-starts/branches-and-loops-local.md [hunk 8]
```diff
@@ -0,0 +1,305 @@
...
+
+## Work with the for loop
+
+Another common loop statement that you'll see in C# code is the
```
Comment (Rick-Anderson) [inline docs/csharp/quick-starts/branches-and-loops-local.md]: The for loop is commonly used in C#.

File: docs/csharp/quick-starts/branches-and-loops-local.md [hunk 9]
```diff
@@ -0,0 +1,305 @@
...
+constructs in the C# language, see if you can write C# code to
+find the sum of all integers 1 through 20 that are divisible
+by 3.  Here are a few hints:
+
+- The `%` operator gives you the remainder of a division operation.
+- The `if` statement givesx you the condition to see if a number should be part of the sum.
+- The `for` loop can help you repeat a series of steps for all the numbers 1 through 20.
+
+Try it yourself. Then check how you did. You can see one possible answer by
+[looking at the finished sample code on GitHub](https://github.com/dotnet/docs/tree/master/samples/csharp/branches-quickstart/Program.cs#L46-L54).
```
Comment (Rick-Anderson) [inline docs/csharp/quick-starts/branches-and-loops-local.md]: `[looking at the finished sample code on GitHub]`
`[viewing the completed code on GitHub]`
or
`[viewing the completed code]`

**Review by Rick-Anderson — APPROVED** [review]
BTW, we discourage NOTE, WARNING, TIP chrome in the asp.net repo.  I've read reports where they are distracting and often skipped. Some repro's like Azure love them.

File: docs/csharp/quick-starts/branches-and-loops.yml [hunk 10]
```diff
@@ -72,7 +73,7 @@ items:
     > The indentation under the `if` and `else` statements is for human readers.
     > The C# language doesn't treat indentation or whitespace as significant. 
     > The statement following the `if` or `else` keyword will be executed based
-    > on the condition. All the samples in this tutorial follow a common
+    > on the condition. All the samples in this quick start follow a common
     > practice to indent lines based on execution.
```
Comment (rpetrusha) [inline docs/csharp/quick-starts/branches-and-loops.yml]: same comment as earlier about "based on execution"

File: docs/csharp/quick-starts/branches-and-loops.yml [hunk 11]
```diff
@@ -268,7 +269,9 @@ items:
     > This online coding experience is in preview mode. If you encounter problems, please report them [on the dotnet/try repo](https://github.com/dotnet/try/issues).
 
   content: |
-    You've completed the "branches and loops" tutorial. You can learn more about these concepts in these topics:
+    You've completed the "branches and loops" quick start. You can learn more about these concepts in these topics:
+
+    You can continue these quick starts on your own development environment. The .NET topic [Get Started in 10 minutes](https://www.microsoft.com/net/core) has instructions for setting up your local development environment. Once you've done that, continue with [List collection quickstart](arrays-and-collections.md) in your local environment.
```
Comment (rpetrusha) [inline docs/csharp/quick-starts/branches-and-loops.yml]: This paragraph interrupts the previous paragraph and the list of topics that's supposed to follow it.

File: docs/csharp/quick-starts/index.md [hunk 12]
```diff
@@ -19,44 +20,52 @@ The first lessons explain C# concepts using small snippets of code. You'll
 learn the basics of C# syntax and how to work with data types like strings,
 numbers, and booleans. It's all interactive, and you'll be writing code
 within minutes. These first lessons assume no prior knowledge of
-programming or the C# language.
+programming or the C# language. 
+
+All the quick starts following the Hello World lesson are available using
+the online browser experience, or on your own development
```
Comment (rpetrusha) [inline docs/csharp/quick-starts/index.md]: no comma after experience

File: docs/csharp/quick-starts/index.md [hunk 13]
```diff
@@ -19,44 +20,52 @@ The first lessons explain C# concepts using small snippets of code. You'll
...
+environment. At the end of each lesson, you decide if you want to continue
+with the next quick start online or on your own machine. There are links
+to help you setup your environment and continue with the next quick start
+on your machine.
 
 ## [Hello world](hello-world.yml)
 
-In the [Hello world](hello-world.yml) lesson, you'll create the most basic
+In the [Hello world](hello-world.yml) quick start, you'll create the most basic
 C#. You'll explore the `string` type and how to work with text.
```
Comment (rpetrusha) [inline docs/csharp/quick-starts/index.md]: C# --> C# program

File: docs/csharp/quick-starts/index.md [hunk 14]
```diff
@@ -19,44 +20,52 @@ The first lessons explain C# concepts using small snippets of code. You'll
...
-mathematical calculations using C#.
+mathematical calculations using C#. This quick start is also available
+[to run locally on your machine](numbers-in-csharp-local.md).
 
-This lesson assumes that you have finished the [Hello world](hello-world.yml) tutorial.
+This quick start assumes that you have finished the [Hello world](hello-world.yml) tutorial.
 
 ## [Branches and loops](branches-and-loops.yml)
 
 The [Branches and loops](branches-and-loops.yml) lesson teaches the basics of selecting
```
Comment (rpetrusha) [inline docs/csharp/quick-starts/index.md]: lesson --> quick start

File: docs/csharp/quick-starts/index.md [hunk 15]
```diff
@@ -19,44 +20,52 @@ The first lessons explain C# concepts using small snippets of code. You'll
...
 different paths of code based on the values stored in variables. You'll learn the
 basics of control flow, which is the basics of how programs make decisions and choose
```
Comment (rpetrusha) [inline docs/csharp/quick-starts/index.md]: basics of how programs --> basis of how programs

File: docs/csharp/quick-starts/list-collection.yml [hunk 16]
```diff
@@ -5,21 +5,19 @@ metadata:
...
   audience: Developer
+  ms.custom: mvc
   level: Beginner
   displayType: two-column
   interactive: csharp
 items:
 - durationInMinutes: 1
   content: |
-    This tutorial teaches you C# interactively, using your browser to write C# and see the results of compiling and running your code. It contains a series of lessons that create, modify and explore collections and arrays.
+    This quick start teaches you C# interactively, using your browser to write C# and see the results of compiling and running your code. It contains a series of lessons that create, modify and explore collections and arrays.
```
Comment (rpetrusha) [inline docs/csharp/quick-starts/list-collection.yml]: write C# --> write C# code

File: docs/csharp/quick-starts/numbers-in-csharp-local.md [hunk 17]
```diff
@@ -0,0 +1,373 @@
...
+ms.topic: article
+ms.prod: .net
+ms.technology: devlang-csharp
+ms.devlang: csharp
+ms.custom: mvc
+---
+
+# Numbers in C# quick start #
+
+This quick start teaches you about the number types in C# interactively. You'll write small amountes of code, then you'll compile and run that code. It contains a series of lessons that explore numbers and math operations in C#. These lessons teach you the fundamentals of the C# language.
```
Comment (rpetrusha) [inline docs/csharp/quick-starts/numbers-in-csharp-local.md]: amountes --> amounts
It contains --> The quick start contains

File: docs/csharp/quick-starts/numbers-in-csharp-local.md [hunk 18]
```diff
@@ -0,0 +1,373 @@
...
+Console.WriteLine(c);
+c = a * b;
+Console.WriteLine(c);
+c = a / b;
+Console.WriteLine(c);
+```
+
+Run this code by typing `dotnet run` in your command window. 
+    
+You can also experiment by performing multiple mathematics operations in the same line, if you'd like.
```
Comment (rpetrusha) [inline docs/csharp/quick-starts/numbers-in-csharp-local.md]: Can you give an example?

File: docs/csharp/quick-starts/numbers-in-csharp-local.md [hunk 19]
```diff
@@ -0,0 +1,373 @@
...
+
+> [!TIP]
+> As you explore C# (or any programming language), you'll
+> make mistakes when you write code. The **compiler** will
+> find those errors and report them to you. When the output
+> contains error messages, look closely at the example code,
```
Comment (rpetrusha) [inline docs/csharp/quick-starts/numbers-in-csharp-local.md]: no comma after code

File: docs/csharp/quick-starts/numbers-in-csharp-local.md [hunk 20]
```diff
@@ -0,0 +1,373 @@
...
+> and the code in the interactive window to see what to fix.
+> That exercise will help you learn the structure of C# code.     
+
+You've finished the first step. Before you start the next section, let's move the current code into a separate method. That makes it easier to start working with a new example. Rename your `Main` method to `WorkWithIntegers` and write a new `Main` method that calls `WorkWithIntegers`. When you have finished, your code should look like this:
```
Comment (rpetrusha) [inline docs/csharp/quick-starts/numbers-in-csharp-local.md]: This is `WorkWithIntegers`, but the code shows `WorkingWithIntegers`.

File: docs/csharp/quick-starts/numbers-in-csharp-local.md [hunk 21]
```diff
@@ -0,0 +1,373 @@
...
+        {
+            WorkingWithIntegers();
+        }
+    }
+}
+```
+
+## Explore order of operations
+
+Comment out the call to `WorkingWithIntegers()`. It will make the output less cluttered as you work in this section:
```
Comment (rpetrusha) [inline docs/csharp/quick-starts/numbers-in-csharp-local.md]: Another WorkingWithIntegers.

File: docs/csharp/quick-starts/numbers-in-csharp-local.md [hunk 22]
```diff
@@ -0,0 +1,373 @@
...
+
+```csharp
+//WorkingWithIntegers();
```
Comment (rpetrusha) [inline docs/csharp/quick-starts/numbers-in-csharp-local.md]: And another WorkingWithIntegers. There are more later, but I won't note them.

File: docs/csharp/quick-starts/numbers-in-csharp-local.md [hunk 23]
```diff
@@ -0,0 +1,373 @@
...
+```
+
+The `//` starts a **comment** in C#. Comments are any text you want to keep in your source code, but not execute as code. The compiler does not generate any executable code from comments.
```
Comment (rpetrusha) [inline docs/csharp/quick-starts/numbers-in-csharp-local.md]: no comma after code.

File: docs/csharp/quick-starts/numbers-in-csharp-local.md [hunk 24]
```diff
@@ -0,0 +1,373 @@
...
+Explore more by combining many different operations. Add something like
+the following lines at the bottom of your `Main` method. Try `dotnet run` again.
+
+```csharp
+d = (a + b) - 6 * c + (12 * 4) / 3 + 12;
+Console.WriteLine(d);
+```
+
+You may have noticed an interesting behavior for integers. Integer
+division always produces an integer result, even when the mathematical
```
Comment (rpetrusha) [inline docs/csharp/quick-starts/numbers-in-csharp-local.md]: even when the mathematical result would... --> even when you'd expect the result to include a decimal or fractional portion.

File: docs/csharp/quick-starts/numbers-in-csharp-local.md [hunk 25]
```diff
@@ -0,0 +1,373 @@
...
+```
+
+The C# integer type differs from mathematical integers in one other
+way: the `int` type has minimum and maximum limits. Add this code to
+your `Main` method to see those limits:
+    
+```csharp
+int max = int.MaxValue;
+int min = int.MinValue;
+Console.WriteLine($"The range of integers is {min} to {max}");
```
Comment (rpetrusha) [inline docs/csharp/quick-starts/numbers-in-csharp-local.md]: This produces numbers that are hard to parse. Decimal separators help: {min:n0} and {max:n0}

File: docs/csharp/quick-starts/numbers-in-csharp-local.md [hunk 26]
```diff
@@ -0,0 +1,373 @@
...
+There are other numeric types with different limits and precision that you
+would use when the `int` type doesn't meet your needs. Let's explore those next.
+
+Once again, let's move the code you wrote in this section into a separate method. Name it `TestLimits`. 
+
+## Work with the double type
+
+The `double` numeric type represents a double-precision floating point
+number. Those terms may be new to you. A **floating point** number is
+useful to represent non-integral numbers that may be very large, or small
```
Comment (rpetrusha) [inline docs/csharp/quick-starts/numbers-in-csharp-local.md]: no comma after large

File: docs/csharp/quick-starts/numbers-in-csharp-local.md [hunk 27]
```diff
@@ -0,0 +1,373 @@
...
+
+```csharp
+double e = 19;
+double f = 23;
+double g = 8;
+double h = (e  + f) / g;
+Console.WriteLine(h);
+```
+
+The range of a double value is much greater than integer values. Try the following
```
Comment (rpetrusha) [inline docs/csharp/quick-starts/numbers-in-csharp-local.md]: Try the following code --> Add the following code

File: docs/csharp/quick-starts/numbers-in-csharp-local.md [hunk 28]
```diff
@@ -0,0 +1,373 @@
...
+code below what you've written so far:
+
+```csharp
+double max = double.MaxValue;
+double min = double.MinValue;
+Console.WriteLine($"The range of double is {min} to {max}");
+```
+
+These values are printed out to represent scientific notation. The number to
```
Comment (rpetrusha) [inline docs/csharp/quick-starts/numbers-in-csharp-local.md]: to represent --> in

File: docs/csharp/quick-starts/numbers-in-csharp-local.md [hunk 29]
```diff
@@ -0,0 +1,373 @@
...
+the left of the `E` is the significant. The number to the right is the exponent,
```
Comment (rpetrusha) [inline docs/csharp/quick-starts/numbers-in-csharp-local.md]: either significand (not significant) or mantissa

File: docs/csharp/quick-starts/numbers-in-csharp-local.md [hunk 30]
```diff
@@ -0,0 +1,373 @@
...
+as a power of 10. Internally, these numbers are represented similarly, but in
```
Comment (rpetrusha) [inline docs/csharp/quick-starts/numbers-in-csharp-local.md]: I think that you're saying that the binary representation is similar to its representation in scientific notation. There are differences, though, and the implication of introducing it here is that it's true for floating-point numbers represented in scientific notation, but not of floating-point numbers not in scientific notation. Personally, I think that this is outside the scope of a beginning topic, so I'd delete it.

File: docs/csharp/quick-starts/numbers-in-csharp-local.md [hunk 31]
```diff
@@ -0,0 +1,373 @@
...
+and division using the `double` type.  Try more complicated calculations.
+
+After you've spent some time with the challenge, take the code you've written
+and place it in a new method. Name that new method `WorkWithDoubles`.
+
+## Work with fixed point types
+
+You've seen the basic numeric types in C#: integers and doubles.  There is one
+other type to learn: the `decimal` type. The `decimal` type has a smaller
+range, but greater precision than `double`. The term **fixed point** means
```
Comment (rpetrusha) [inline docs/csharp/quick-starts/numbers-in-csharp-local.md]: no comma after range

File: docs/csharp/quick-starts/numbers-in-csharp-local.md [hunk 32]
```diff
@@ -0,0 +1,373 @@
...
+
+Notice that the math using the decimal type has more digits to the right
+of the decimal point. 
+
+***Challenge***
+
+Now that you've seen the different numeric types, write code that calculates
+the area of a circle whose radius is 2.50 inches. Remember that the area of a circle
+is the radius squared multiplied by PI. One hint: C# contains a constant
+for PI: `Math.PI` that you can use for that value.
```
Comment (rpetrusha) [inline docs/csharp/quick-starts/numbers-in-csharp-local.md]: PI: `Math.PI` that --> PI, `Math.PI`, that
possibly use a link (\<xref:System.Math.PI?displayProperty=nameWithType>)?

**Review by rpetrusha — CHANGES_REQUESTED** [review]
I've left a number of comments and suggestions, @BillWagner.

File: docs/csharp/quick-starts/branches-and-loops-local.md [hunk 33]
```diff
@@ -0,0 +1,305 @@
...
+
+# Branches and loops
+
+This quick start teaches you about how to write code that examines variables and changes execution path based on those variables. You'll write C# and see the results of compiling and running your code. It contains a series of lessons that explore branching and looping constructs in C#. These lessons teach you the fundamentals of the C# language.
+
+This quick start expects you to have a machine you can use for development. The .NET topic [Get Started in 10 minutes](https://www.microsoft.com/net/core) has instructions for setting up your local development environment on Mac, PC or Linux.
+
+## Make decisions using the `if` statement
+
+Create a directory named **branches-quickstart**. Make that the current directory and run `dotnet new console -n BranchesAndLoops -o .`.
```
[the remark this answers is not in the document]
Comment (BillWagner) [reply docs/csharp/quick-starts/branches-and-loops-local.md]: We've been always creating the directory as a separate step. I did add an explanation of the command.

File: docs/csharp/quick-starts/branches-and-loops-local.md [hunk 34]
```diff
@@ -0,0 +1,305 @@
...
+```
+
+The statement following the `else` keyword executes only when the condition being tested is `false`. Combining `if` and `else` with boolean conditions provides all the power you need.
+
+> [!IMPORTANT]
+> The indentation under the `if` and `else` statements is for human readers.
+> The C# language doesn't treat indentation or whitespace as significant. 
+> The statement following the `if` or `else` keyword will be executed based
+> on the condition. All the samples in this quick start follow a common
+> practice to indent lines based on execution.
```
[the remark this answers is not in the document]
Comment (BillWagner) [reply docs/csharp/quick-starts/branches-and-loops-local.md]: I tried the phrase "control flow of statements".

File: docs/csharp/quick-starts/branches-and-loops-local.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (BillWagner) [reply docs/csharp/quick-starts/branches-and-loops-local.md]: Rick had the same comment. Changed.

File: docs/csharp/quick-starts/branches-and-loops.yml (hunk 10 above)
Comment (BillWagner) [reply docs/csharp/quick-starts/branches-and-loops.yml]: I tried the same change here.

File: docs/csharp/quick-starts/index.md (hunk 15 above)
Comment (BillWagner) [reply docs/csharp/quick-starts/index.md]: nice suggestion!

File: docs/csharp/quick-starts/numbers-in-csharp-local.md (hunk 20 above)
Comment (BillWagner) [reply docs/csharp/quick-starts/numbers-in-csharp-local.md]: I went with the code as correct. :)

File: docs/csharp/quick-starts/numbers-in-csharp-local.md (hunk 24 above)
Comment (BillWagner) [reply docs/csharp/quick-starts/numbers-in-csharp-local.md]: Good comment. I also changed the YML version that had the same text.

File: docs/csharp/quick-starts/numbers-in-csharp-local.md (hunk 25 above)
Comment (BillWagner) [reply docs/csharp/quick-starts/numbers-in-csharp-local.md]: I left that out on purpose, because I haven't explained format strings yet. But I agree with your comment.

File: docs/csharp/quick-starts/numbers-in-csharp-local.md (hunk 30 above)
Comment (BillWagner) [reply docs/csharp/quick-starts/numbers-in-csharp-local.md]: good point.

**Review by rpetrusha — APPROVED** [review]

**Comment by rpetrusha** [comment]
LGTM, @BillWagner. This is ready to merge when you want.
